### PR TITLE
Signal navigation

### PR DESCRIPTION
**Pull request by SuicideBunny8**
Works in theory. I have no way of testing as of right now

### CODE REVIEW

File: BaseStation/BaseServer/BaseServer.cs [hunk 1]
```diff
@@ -196,6 +199,24 @@ private static List<float> ConvertToFloatArray(Packet data)
             return ret;
         }
 
+        private static List<double> ConvertToDoubleArray(Packet data)
```
Comment (Obbay2) [inline BaseStation/BaseServer/BaseServer.cs]: All instances of use of ConvertToFloatArray should be removed and replaced with this? This is basically correct duplicated code and code that used ConvertToFloatArray should be able to cast down.

Comment (SuicideBunny8) [reply BaseStation/BaseServer/BaseServer.cs]: this one divides it into arrays of length 8 which are needed for doubles. you would be losing half of your data if you tried to use the this version of the converting on a string of floats. it's a necessary duplication.

Comment (SuicideBunny8) [reply BaseStation/BaseServer/BaseServer.cs]: it's possible to factor out some of it however

Comment (Obbay2) [reply BaseStation/BaseServer/BaseServer.cs]: You can factor out basically all of it and just make a second parameter with how many values to unpack!

Comment (SuicideBunny8) [reply BaseStation/BaseServer/BaseServer.cs]: I can't factor out the utildata.tofloat and untildata.todouble

**Pull request merged**